regolith/helpers/makeappointmentshelper.py

```diff
@@ -133,137 +158,101 @@ def sout(self):
+                                   f" grant end: {all_grants[grant]['end_date']} budget end: {budget_end}")
+            days_to_go = (all_grants[grant]['end_date'] - projection_from_date).days
+            this_burn = all_grants[grant]['burn']
+            end_amount = this_burn.get(all_grants[grant]['end_date'])['student_days'] + \
+                         this_burn.get(all_grants[grant]['end_date'])['ss_days'] + \
+                        this_burn.get(all_grants[grant]['end_date'])['postdoc_days']
             if end_amount > 15.25:
-                underspent.append("    end: {}, grant: {}, underspend amount: {} months,\n      required ss+gra burn: {}".
-                    format(str(grant_end), grant.get('alias'), round(end_amount/30.5, 2),
+                underspent.append("    end: {}, grant: {}, underspend amount: {} months,\n"
+                                  "      required ss+gra burn: {}".
+                    format(str(all_grants[grant]['end_date']), grant, round(end_amount/30.5, 2),
                     round(end_amount / days_to_go, 2)))
                 cum_underspend += end_amount
             elif end_amount < -30.5:
                 overspent.append("    end: {}, grant: {}, overspend amount: {} months".format(
-                    str(grant_end), grant.get('alias'), round(end_amount/30.5, 2)))
+                    str(all_grants[grant]['end_date']), grant, round(end_amount/30.5, 2)))
+            # values for individual and cumulative grant burn plots
             if not rc.no_plot:
-                grant_dates = []
-                grant_duration = (grant_end - grant_begin).days + 1
-                this_student, this_pd, this_ss = [0.0] * grant_duration, [
-                    0.0] * grant_duration, [0.0] * grant_duration
+                grant_dates = [all_grants[grant]['begin_date'] + relativedelta(days=x) for x in
+                              range((all_grants[grant]['end_date'] - all_grants[grant]['begin_date']).days + 1)]
+                this_student, this_pd, this_ss = [0.0] * len(grant_dates), [0.0] * len(grant_dates), \
+                                                 [0.0] * len(grant_dates)
                 counter = 0
                 for x in range(len(datearray)):
-                    if is_current(grant, now=datearray[x]):
-                        grant_dates.append(datearray[x])
-                        this_student[counter] = grant_amounts[counter].get(
-                            'student_days')
-                        cum_student[x] += grant_amounts[counter].get('student_days')
-                        this_pd[counter] = grant_amounts[counter].get(
-                            'postdoc_days')
-                        cum_pd[x] += grant_amounts[counter].get('postdoc_days')
-                        this_ss[counter] = grant_amounts[counter].get('ss_days')
-                        cum_ss[x] += grant_amounts[counter].get('ss_days')
+                    if day_burn := this_burn.get(datearray[x]):
```

Comment:
I have used the controversial walrus operator here... can remove if not readable

Reply:
its ok, except that it pins all of regolith to python 3.8.....  We could wrap it in a statement that checks the version of python, or go with a conventional statement.  It is mostly syntactic sugar here.

Reply:
best to remove it then!

Reply:
its gonna be cool, but having fun has consequences..... :)

Reply:
removed :(